lib/OpenQA/WebAPI/Controller/API/V1.pm

```diff
@@ -49,16 +49,26 @@ sub auth {
             #$self->app->log->debug("$hash $msg");
             my $timestamp          = $headers->header('X-API-Microtime');
             my $build_tx_timestamp = $headers->header('X-Build-Tx-Time');
+            my $username           = $api_key->user->username;
+            my $request_ip         = $headers->header("x-forwarded-for") || "unknown";
             if ($self->_valid_hmac($hash, $msg, $build_tx_timestamp, $timestamp, $api_key)) {
                 $user = $api_key->user;
             }
             else {
-                $self->app->log->error("hmac check failed");
+                my $log_msg
+                  = sprintf "Rejecting authentication for user '%s' with ip '%s'. Valid key '%s', secret '%s'",
+                  $username, $request_ip, $api_key->key, $api_key->secret;
                 if (!_is_timestamp_valid($build_tx_timestamp, $timestamp)) {
                     $reason = "timestamp mismatch";
+                    $self->app->log->warn($log_msg . ", $reason");
```

Comment:
Style nitpick: Since there are so many calls to `$self->app->log` in this method, it would make sense to assign the object to a variable somewhere at the top (`my $log = $self->app->log;`). Method calls in Perl are rather expensive, maybe not enough to make much of a difference here, but it's good practice.